packages/storage-plus/src/map.rs

```diff
@@ -308,6 +318,32 @@ mod test {
                 (b"recipient2".to_vec(), 3000)
             ]
         );
+
+        // let's iterate over a sub prefix
+        let all: StdResult<Vec<_>> = TRIPLE
+            .sub_prefix(b"owner")
+            .range(&store, None, None, Order::Ascending)
+            .collect();
+        let all = all.unwrap();
+        assert_eq!(3, all.len());
+        // FIXME: range() works, but remaining keys are still encoded
+        assert_eq!(
+            all,
+            vec![
+                (
+                    [to_length_prefixed(b"\x09"), b"recipient".to_vec()].concat(),
```

Comment:
Ah, the big-endian encoding of U8key manually...
This is fine for now. It would be interesting to see how to auto-parse these keys, but the caller should know how in the end

Reply:
Yes. It could be handled automatically by the deserialization function (returning a vec! of (properly deserialized) tuples). But, for that we need to know if the sub-prefix is composed by one or more keys... if all the keys where length-prefixed that would be straightforward: we just honour the length-prefixed information. But currently, the last key is not length-prefixed. So, this introduces ambiguity.

I think an easy fix would be to make all keys length-prefixed, and for the deserilization fn to handle that directly.

Reply:
We could also consider reducing the max length of length-prefixed keys to one byte (255), to save some bytes / processing.